### ISSUE TITLE
creatPostHandler: better error messages

### ISSUE THREAD

**Issue by houssemAmiri** [yebrahim/codersquare#21]

**Comment by yebrahim** [comment]
agreed, I think we could use some review of all error messages, and maybe better type-checking as well (maybe add an enum for errors?).

**Comment by houssemAmiri** [comment]
I'm thinking about how to do it nest js have good error handling I'm trying to understand how it work and implement it here I will open a PR next days

**Comment by yebrahim** [comment]
can you explain more what you have in mind? I thought you just meant going over the error strings we're returning to make sure they're useful.

**Comment by houssemAmiri** [comment]
I suggest that we add using this library 
https://express-validator.github.io/docs/ 
it's a middleware that wraps errors 
also adding enum for error types 
and classes to initiate common errors 
example: 
`new HttpException('Forbidden', HttpStatus.FORBIDDEN);`

````
export class ForbiddenException extends HttpException {
  constructor() {
    super('Forbidden', HttpStatus.FORBIDDEN);
  }
}
````
and 
`throw new ForbiddenException();`
what do you think @yebrahim we can do this fo all errors

**Comment by yebrahim** [comment]
it's a good find.
the problem with these libraries is they're pretty opinionated, and they end up shaping your code in a way you don't necessarily like, and now newcomers to the codebase have to learn to use those and commit to them as well.
I think for a small project like this, we're better served rolling our own verification utilities, which won't be hard to do anyway.
let me know if you feel strongly about this though.
